### PR TITLE
Favorite features

### PR DESCRIPTION
**Pull request by whitneykidd**
Completed user stories 12 - 15
includes functionality for stories 1 - 15

### CODE REVIEW

File: app/controllers/favorite_controller.rb [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+    @favorites = favorite.favorite_pets
+  end
+
   def update
     pet = Pet.find(params[:pet_id])
-    favorite.add_pet(pet.id)
+    favorite.add_pet?(pet.id)
     session[:favorite] = favorite.contents
-    quantity = favorite.count_of(pet.id)
+    # quantity = favorite.count_of(pet.id)
```
Comment (whitneykidd) [inline app/controllers/favorite_controller.rb:10]: Not sure if we still need this, it was created from following the lesson plan

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -13,7 +13,7 @@
       <center>Adopt Don't Shop
       <%= link_to "All Pets", "/pets" %>
       <%= link_to "All Shelters", "/shelters" %>
-      <p>Favorite Pets: <%= favorite.total_count %></p>
+      <%= link_to "Favorite Pets: #{favorite.total_count}", "/favorites" %>
```
Comment (whitneykidd) [inline app/views/layouts/application.html.erb:16]: Updated to link as per user story

File: app/assets/stylesheets/application.css [hunk 3]
```diff
@@ -130,3 +130,7 @@ label {
 .column {
   flex: 50%;
 }
+
+.fav_pet {
```
Comment (janegreene) [inline app/assets/stylesheets/application.css:134]: Looks nice, thank you for handling this!

File: app/controllers/favorite_controller.rb [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
+   if params[:page] == "favorites"
+      redirect_to '/favorites'
+    elsif params[:page] == "show"
+      flash[:notice] = "You have removed #{pet.name} from Favorite Pets."
+      redirect_to "/pets/#{pet.id}"
+    end
   end
 
+  def destroy_all
+    reset_session
```
Comment (janegreene) [inline app/controllers/favorite_controller.rb:28]: very crafty

**Pull request merged**